### PR TITLE
[RPS-132] Prevent saving publication with blank Granularity field

### PR DESCRIPTION
**Pull request by ziemsky**

### CODE REVIEW

File: acceptance-tests/src/test/java/uk/nhs/digital/ps/test/acceptance/pages/GranularitySection.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    public void addGranularityField() {
+        helper.executeWhenStable(() -> findAddButton().click());
+    }
+
+    private WebElement findRootElement() {
+        // find 'parent' div that _contains_ an 'h3/span' child elements, where the <span> has text 'Granularity'
+        // this is so that further searches can be performed in context of the root element
+        return getWebDriver().findElement(
+            By.xpath("//div[@class='hippo-editor-field' and h3/span[text() = 'Granularity']]"));
```
Comment (wojtek-oledzki) [inline acceptance-tests/src/test/java/uk/nhs/digital/ps/test/acceptance/pages/GranularitySection.java]: `XpathSelectors.EDITOR_BODY + ` to ensure that we are selecting the field from current, active editor

File: repository-data/application/src/main/resources/hcm-config/configuration/frontend/cms.yaml [hunk 2]
```diff
@@ -24,3 +24,6 @@ definitions:
     /hippo:configuration/hippo:frontend/cms/cms-validators/blank-attachment-field-forbidden:
       jcr:primaryType: frontend:plugin
       plugin.class: uk.nhs.digital.ps.BlankAttachmentFieldValidator
+    /hippo:configuration/hippo:frontend/cms/cms-validators/blank-granularity-forbidden:
```
Comment (wojtek-oledzki) [inline repository-data/application/src/main/resources/hcm-config/configuration/frontend/cms.yaml]: #26 brings in some namespacing for validators. We should start isolating PS code and config a bit more.

Comment (ziemsky) [reply repository-data/application/src/main/resources/hcm-config/configuration/frontend/cms.yaml]: I'll wait for #26.

File: cms/src/test/java/uk/nhs/digital/ps/BlankGranularitySelectionFieldValidatorTest.java [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    private IFieldValidator fieldValidator;
+    @Mock private IFieldDescriptor fieldDescriptor;
+    @Mock private ITypeDescriptor typeDescriptor;
+    @Mock private JcrNodeModel documentNodeModel;
+    @Mock private JcrPropertyValueModel fieldModel;
+    @Mock private IPluginConfig pluginConfig;
+    @Mock private IPluginContext pluginContext;
+    @Mock private IModel<String> violationMessageTranslationModel;
+
+    private BlankGranularitySelectionFieldValidator blankAttachmentFieldValidator;
```
Comment (ngta1) [inline cms/src/test/java/uk/nhs/digital/ps/BlankGranularitySelectionFieldValidatorTest.java]: Rename variable to "blankGranularitySelectionFieldValidator"?

Comment (ziemsky) [reply cms/src/test/java/uk/nhs/digital/ps/BlankGranularitySelectionFieldValidatorTest.java]: Good spot! I'll rename the field.

Comment (ziemsky) [reply cms/src/test/java/uk/nhs/digital/ps/BlankGranularitySelectionFieldValidatorTest.java]: Done.

File: acceptance-tests/src/test/java/uk/nhs/digital/ps/test/acceptance/pages/GranularitySection.java (hunk 1 above)
Comment (ziemsky) [reply acceptance-tests/src/test/java/uk/nhs/digital/ps/test/acceptance/pages/GranularitySection.java]: Done.

File: repository-data/application/src/main/resources/hcm-config/configuration/frontend/cms.yaml (hunk 2 above)
Comment (ziemsky) [reply repository-data/application/src/main/resources/hcm-config/configuration/frontend/cms.yaml]: Now rebased.

**Comment by ziemsky** [comment]
Changes resulting from the comments have been applied and branch rebased to pull latest code from `master`.

Note that the validator has now also been made more generic, so, from now on, rather than targeting the 'Granularity' field specifically, it can be used for all fields of type 'StaticDropdown'.

**Pull request merged**